CCDB/src/CcdbApi.cxx

```diff
@@ -97,6 +100,7 @@ std::unique_ptr<std::vector<char>> CcdbApi::createObjectImage(const void* obj, s
 {
   // Create a binary image of the object, if CcdbObjectInfo pointer is provided, register there
   // the assigned object class name and the filename
+  std::lock_guard<std::mutex> guard(gIOMutex);
```

Comment:
I am wonderning, why do you need to make CcdbApi threadsafe? This class is not a singleton and one could easily have a CccbApi instance per thread. At first, I understood that thread-safety is attacked via BasicCCDBManager but maybe this is not the case?

I am open to these changes but wonder if we need to pay the price of mutexes all the time.

Reply:
I agree with Sandro. Although speed is not the most relevant feature of the CcdbApi (it is slow due to protocol), I think that it was designed with the idea that you can have as many instances as you need. In QC we have 2 for example, one for the QCDB and one for the CCDB.